### PR TITLE
High risk client categorization

### PR DESCRIPTION
**Pull request by gabriel090**
PMTC High risk client categorization

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/HighRiskClientCategorization.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.Map;
+import java.util.Set;
+
+public class HighRiskClientCategorization extends AbstractPatientCalculation implements PatientFlagCalculation {
+    protected static final Log log = LogFactory.getLog(HighRiskClientCategorization.class);
+    @Autowired
+    private EncounterService encounterService;
+    @Override
+    public String getFlagMessage() {
+        return "High risk client categorization";
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/HighRiskClientCategorization.java:36]: Shouldn't the flag only read High risk client?

File: api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/HighRiskClientCategorization.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            boolean result = false;
+            Patient patient = patientService.getPatient(ptId);
+            Double lastVlResultValue = null;
+            CalculationResult lastvlresult = lastVlResults.get(ptId);
+            Date transferInDateValue = EmrCalculationUtils.datetimeResultForPatient(transferInDate, ptId);
+            EncounterType mchEnrollment = MetadataUtils.existing(EncounterType.class, MchMetadata._EncounterType.MCHMS_ENROLLMENT);
+            Form antenatalVisitForm = MetadataUtils.existing(Form.class, MchMetadata._Form.MCHMS_ANTENATAL_VISIT);
+            Encounter mchMsEnc = EmrUtils.lastEncounter(patient,mchEnrollment,antenatalVisitForm );
+
+            if (inHivProgram.contains(ptId)) {
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/HighRiskClientCategorization.java:72]: I guess you need some sort of condition nesting. Please check

**Review by ojwanganto — CHANGES_REQUESTED** [review]

**Comment by gabriel090** [comment]
Ok. I will work on it and consult more

**Review by patryllus — COMMENTED** [review]
All new HIV positives irrespective of time identified (including those seroconverted )


You missed out this.

**Pull request closed without merging**